### PR TITLE
CorrDiff: inference bugfixes, cleanup, and documentation improvements

### LINKED ISSUE #860: 📚[DOC]: CorrDiff Validation and Early Stopping
### How would you describe the priority of this documentation request

Medium

### Is this for new documentation, or an update to existing docs?

Update

### Describe the incorrect/future/missing documentation

In CorrDiffs ```train.py``` there is support for logging validation loss based on the presense of a validation key in the training config yaml. In [dataset.py](https://github.com/NVIDIA/physicsnemo/blob/ad863fdb8ff2e263551e4181b7799f79a4c5177d/examples/generative/corrdiff/datasets/dataset.py#L121) it looks like the training dataset configuration is copied for repurposing for validation with the option to update the configuration. I can see how one would provide a secondary path to a separate dataset used for validation, which would replace the dataset.data_path in either the training config (https://github.com/climacell/physicsnemo/blob/main/examples/generative/corrdiff/conf/config_training_hrrr_mini_regression.yaml) or via hydra override. Is this the intended validation process? I am confused by the```test_train_split``` boolean used to initiate the validation flow if the validation is done on a separate dataset and not actaully split out from the training dataset. 

Additionally, if validation logging is supported, is there configurable early stopping for the regression? 

Thanks for clarity.

### PR DESCRIPTION
**Pull request by CharlelieLrt**
# PhysicsNeMo Pull Request

## Description
This PR addresses multiple bugs related to CorrDiff training and inference:
- It improves clarity regarding (cross-)validation through documentation and variable renaming.
- It removes duplicate code from `ResidualLoss`
- Automatically disables AMP for inference in `generate.py` (bug reported in #874).
- Implements no-ops fallbacks for profiling tools when torch is built without cuda support (useful to run hrrr-mini on CPU)

Closes #860.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

## Dependencies

None.

### CODE REVIEW

**Review by NickGeneva — APPROVED** [review]
LGTM, nice fixes
